spec/unit/resources/service_spec.rb

```diff
@@ -0,0 +1,140 @@
+require_relative '../spec_helper'
+include CernerSplunk::PathHelpers, CernerSplunk::PlatformHelpers, CernerSplunk::ServiceHelpers, CernerSplunk::RestartHelpers
+
+describe 'splunk_service' do
+  let(:runner_params) { { platform: 'redhat', version: '7.1' } }
+  let(:test_params) { { name: 'splunk', action: :nothing } }
+
+  let(:mock_run_state) { { 'splunk_ingredient' => { 'installations' => {} } } }
+  let(:chef_run) do
+    ChefSpec::SoloRunner.new({ step_into: ['splunk_service'] }.merge!(runner_params)) do |node|
+      node.normal['test_parameters'] = test_params
+      node.normal['run_state'].merge!(mock_run_state)
+    end.converge('cerner_splunk_ingredient_test::service_unit_test')
+  end
+
+  let(:ftr) { double('ftr_pathname') }
+  let(:init_script) { double('init_script_path') }
+
+  let(:run_state) { chef_run.node.run_state['splunk_ingredient'] }
+
+  %w(start stop restart).each do |action|
+    describe "with action :#{action}" do
+      platform_package_matrix.select { |k, _| %w(windows redhat).include? k }.each do |platform, versions|
+        versions.each do |version, packages|
+          describe "on #{platform} #{version}" do
+            packages.each do |package, _|
+              context "with #{package}" do
+                let(:runner_params) { { platform: platform, version: version, user: 'root' } }
+                let(:test_params) { { name: package.to_s, action: action.to_sym } }
+                let(:install_dir) { default_install_dirs[package][is_windows ? :windows : :linux] }
```

Comment:
Are all of these let blocks needed?  I'm having a hard time finding where install_dir and cwd are used.